app/controllers/pieces_controller.rb

```diff
@@ -7,17 +10,30 @@ def show
 
   def update
     @selected_piece = ChessPiece.find(params[:id])
+    @game = @selected_piece.game
 
-    if @selected_piece && @selected_piece.valid_move?(Coordinates.new(move_to_x_parameter, move_to_y_parameter))
+    if moving_validly?
       @selected_piece.move_to!(Coordinates.new(move_to_x_parameter, move_to_y_parameter))
```

Comment:
Does `move_to` call `valid_move?` already?

Wondering if we are calling `valid_move?` twice here.